test/trackForMutations.spec.js

```diff
@@ -18,7 +18,27 @@ describe('trackForMutations', () => {
   function testCasesForNonMutation(spec) {
     it('returns false', () => {
       const state = spec.getState();
-      const tracker = trackForMutations(isImmutable, state);
+      const tracker = trackForMutations(isImmutable, undefined, state);
+      const newState = spec.fn(state);
+
+      expect(
+        tracker.detectMutations()
+      ).toEqual({wasMutated: false});
+    });
+  }
+
+  function testCasesForIgnoreMutationDetectionByPath(spec) {
```

Comment:
gotcha. I definitely like that better, especially once the middleware is created with a single `options` argument (instead of `isImmutable` and `options`).